applications/EigenSolversApplication/custom_solvers/feast_eigensystem_solver.h

```diff
@@ -0,0 +1,435 @@
+
+        // truncate non-converged results
+        tmp_eigenvalues.resize(M, true);
+        tmp_eigenvectors.resize(system_size, M, true);
+
+        RetrieveEigensolution<OutputVectorType, VectorType>(tmp_eigenvalues, rEigenvalues, tmp_eigenvectors, rEigenvectors);
+    }
+
+    DenseOutputMatrixType& GetEigenvectorSolution()
+    {
+        KRATOS_ERROR_IF( mpEigenvectorMatrix == nullptr ) << "The eigenvector matrix is not initialized. This means, that the real general generalized " <<
+            "eigenvalue problem did not yield complex results or the solution has not yet been performed" << std::endl;
+        return *mpEigenvectorMatrix;
+    }
+
+    OutputVectorType& GetEigenvalueSolution()
+    {
+        KRATOS_ERROR_IF( mpEigenvectorMatrix == nullptr ) << "The eigenvector matrix is not initialized. This means, that the real general generalized " <<
+            "eigenvalue problem did not yield complex results or the solution has not yet been performed" << std::endl;
+        return *mpEigenvalueVector;
+    }
+
+    /**
+     * Print information about this object.
+     */
+    void PrintInfo(std::ostream &rOStream) const override
+    {
+        rOStream << "FEASTEigensystemSolver";
+    }
+
+    /**
+     * Print object's data.
+     */
+    void PrintData(std::ostream &rOStream) const override
+    {
+    }
+
+  private:
+
+    typename TDenseSpaceTypeOut::MatrixPointerType mpEigenvectorMatrix;
+
+    typename TSparseSpaceTypeOut::VectorPointerType mpEigenvalueVector;
+
+    typedef void (feast_ptr)(char*, int*, double*, int*, int*, double*, int*, int*, int*, double*, int*, double*, double*, int*, double*, double*, int*, double*, int*);
+
+    std::function<feast_ptr> CallFeast(double T, bool symmetric)
+    {
+        using namespace std::placeholders;
+
+        if( symmetric ) {
+            return std::bind(dfeast_scsrgv, _1, _2, _3, _4, _5, _6, _7, _8, _9, _10, _11, _12, _13, _14, _15, _16, _17, _18, _19);
+        } else {
+            return std::bind(dfeast_gcsrgv, _2, _3, _4, _5, _6, _7, _8, _9, _10, _11, _12, _13, _14, _15, _16, _17, _18, _19);
+        }
+    }
+
+    std::function<feast_ptr> CallFeast(std::complex<double> T, bool symmetric)
+    {
+        using namespace std::placeholders;
+
+        if( symmetric ) {
+            return std::bind(zfeast_scsrgv, _1, _2, _3, _4, _5, _6, _7, _8, _9, _10, _11, _12, _13, _14, _15, _16, _17, _18, _19);
+        } else {
+            return std::bind(zfeast_gcsrgv, _2, _3, _4, _5, _6, _7, _8, _9, _10, _11, _12, _13, _14, _15, _16, _17, _18, _19);
+        }
+    }
+
+    double GetE1(double T)
+    {
+        return mParam["e_min"].GetDouble();
+    }
+
+    double GetE2(double T)
+    {
+        return mParam["e_max"].GetDouble();
+    }
+
+    std::complex<double> GetE1(std::complex<double> T)
+    {
+        return std::complex<double>(mParam["e_mid_re"].GetDouble(), mParam["e_mid_im"].GetDouble());
+    }
+
+    double GetE2(std::complex<double> T)
```

Comment:
in those functions `T` is never used
Edit: Ok in some but I think can be done cleaner
EditEdit: Reading the rest of the Code I think you need to do this for when it is complex right?
Cannot you specialize the function based on the templates of the Class? then you don't need this auxiliary `T` parameter

Reply:
I tried but didn't get it to work yet. I will try harder

Reply:
I now used `enable_if` to get rid of the dummy parameter